lib/lintPostcssResult.js

```diff
@@ -107,16 +107,27 @@ function lintPostcssResult(stylelintOptions, postcssResult, config) {
 
 		performRules.push(
 			Promise.all(
-				postcssRoots.map((postcssRoot) =>
-					ruleFunction(primaryOption, secondaryOptions, {
-						fix:
-							stylelintOptions.fix &&
-							// Next two conditionals are temporary measures until #2643 is resolved
-							isFileFixCompatible &&
-							!postcssResult.stylelint.disabledRanges[ruleName],
-						newline,
-					})(postcssRoot, postcssResult),
-				),
+				postcssRoots.map(async (postcssRoot) => {
+					try {
+						return await ruleFunction(primaryOption, secondaryOptions, {
+							fix:
+								stylelintOptions.fix &&
+								// Next two conditionals are temporary measures until #2643 is resolved
+								isFileFixCompatible &&
+								!postcssResult.stylelint.disabledRanges[ruleName],
+							newline,
+						})(postcssRoot, postcssResult);
+					} catch (ex) {
+						const fileName =
+							postcssRoot.source && postcssRoot.source.input
+								? postcssRoot.source.input.file
+								: 'unknown file';
```

Comment:
When I test this change with `--stdin`, the message includes `undefined` instead of `unknown file`:

```console
$ cat a.sass | bin/stylelint.js --stdin --syntax=sass
Rule indentation threw an unknown exception from undefined
```

So, can we add test cases about this change? It seems good to me to add them to `lib/__tests__/standalone.test.js`.